os_migrate/plugins/module_utils/resource.py

```diff
@@ -33,10 +33,17 @@ class Resource():
     # Items of refs dict that are used as SDK call kwargs when
     # issuing a create/update REST API request.
     sdk_params_from_refs = []
-    # some parameters are allowed when creating a resource but not when
+    # Some parameters are allowed when creating a resource but not when
     # updating it.  This list of parameter names is purged from the parameter
     # list before calling update.
     readonly_sdk_params = []
+    # Some parameters get serialized as e.g. an empty array and it can
+    # matter whether they are fed into a create/update request or
+    # skipped altogether. Parameters listed here will not be included
+    # in sdk_params in case their value is falsey - false, empty
+    # array, empty dict etc. (Parameters with None value are skipped
+    # anyway and do not need to be listed here.)
+    skip_falsey_sdk_params = []
```

Comment:
I think we're gradually getting to a point where instead of having extra lists like `readonly_sdk_params` and `skip_falsey_sdk_params`, we should probably refactor the parameter lists to be parameter dictionaries, and allow them to carry info about how the parameter should be treated. E.g.:

```python
params_from_sdk = {
    'availability_zone_hints': { 'skip_falsey': true },
    'some_normal_param': {},
    'some_readonly_param': { 'readonly': true },
}
```

I think that would be a separate refactoring task though.